drive/amazon/amazon.go

```diff
@@ -46,29 +47,30 @@ func NewClient(c drive.Config) (drive.Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &AmazonCloudDrive{
+	return &Drive{
```

Comment:
I got to the bottom of your pull request, and I now see you're proposing they are *all* named just Drive.  I did some quick testing to reassure myself that Go always prints the package name when printing the type, and as such it's fine.  Please disregard this set of comments.